File: api/src/main/java/dev/obscuria/fragmentum/content/util/easing/CubicCurve.java
package dev.obscuria.fragmentum.content.util.easing;

import org.apache.commons.lang3.NotImplementedException;

@SuppressWarnings("unused")
public final class CubicCurve
{
    public CubicCurve(int resolution) {}

    public CubicCurve addPoint(float x, float y)
    {
        throw new NotImplementedException();
    }

    public CubicCurve addPoint(double x, double y)
    {
        throw new NotImplementedException();
    }

    public Point samplePoint(float x)
    {
        throw new NotImplementedException();
    }

    public float sample(float x)
    {
        throw new NotImplementedException();
    }

    public EasingFunction toEasing()
    {
        throw new NotImplementedException();
    }

    public record Point(float x, float y)
    {
        public static final Point ZERO = new Point(0, 0);
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/tools/easing/CubicCurve.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.tools.easing

class CubicCurve(val resolution: Int) {

    fun addPoint(x: Float, y: Float): CubicCurve = error("API stub")

    fun addPoint(x: Double, y: Double): CubicCurve = error("API stub")

    fun samplePoint(x: Float): Point = error("API stub")

    fun sample(x: Float): Float = error("API stub")

    fun asEasing(): EasingFunction = error("API stub")

    data class Point(val x: Float, val y: Float) {

        companion object {

            val ZERO: Point = Point(0f, 0f)
        }
    }
}